thold_functions.php

```diff
@@ -2348,23 +2353,47 @@ function thold_check_threshold(&$thold_data) {
 			db_execute_prepared('UPDATE thold_data
 				SET thold_alert = ?,
 				thold_fail_count = ?,
-				thold_warning_fail_count = 0
+				thold_warning_fail_count = ?
 				WHERE id = ?',
-				array($thold_data['thold_alert'], $thold_data['thold_fail_count'], $thold_data['id']));
+				array($thold_data['thold_alert'], $thold_data['thold_fail_count'], $thold_data['thold_warning_fail_count'], $thold_data['id']));
+
+		/* Warning status */
```

Comment:
This really does not accomplish anything.  Please explain why you are changing the behavior here.  If figure it does not hurt.  Was there something 'upstream' needed with the 'thold_fail_count'?